### PR TITLE
Add flake metadata

### PR DESCRIPTION
**Pull request by thefloweringash**

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+  inputs.branchctl.url = "git+ssh://git@github.com/iknow/branchctl?ref=flakes";
+  inputs.branchctl.inputs.nixpkgs.follows = "nixpkgs";
+  inputs.branchctl.inputs.flake-utils.follows = "flake-utils";
+
+  outputs = { self, nixpkgs, flake-utils, branchctl }:
+    let
+      supportedSystems = [ "x86_64-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ];
+      overlay = import ./overlays { inherit branchctl; };
```
Comment (thefloweringash) [inline flake.nix]: I'd like to drop the duplicated branchctl packages and use
```nix
{
  overlay = nixpkgs.lib.composeExtensions branchctl.overlay (import ./overlays);
}
```

But I don't see a nice way to do this while maintaining support for non-flake systems.

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    in
+  {
+    inherit overlay;
+    nixosModule = {
+      imports = [ ./nixos-modules/all-modules.nix ];
+      nixpkgs.overlays = [ overlay ];
```
Comment (thefloweringash) [inline flake.nix:22]: I don't know if this is sensible in the land of flakes, but it's consistent with the current module.

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{
+  description = "iKnow's shared nix utilities";
+
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+  inputs.branchctl.url = "git+ssh://git@github.com/iknow/branchctl?ref=flakes";
```
Comment (thefloweringash) [inline flake.nix]: Reminder for self: drop `?ref=flakes` before merging. (See https://github.com/iknow/branchctl/pull/7, https://github.com/iknow/branchctl/pull/13).

**Review by thatsmydoing — APPROVED** [review]
LGTM

**Pull request merged**